backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/predicate/impl/AbstractElasticsearchSearchPredicateBuilder.java

```diff
@@ -48,7 +62,31 @@ public final JsonObject build(ElasticsearchSearchPredicateContext context) {
 		}
 
 		JsonObject result = doBuild( context, outerObject, innerObject );
-		return ( withConstantScore ) ? applyConstantScore( result ) : result;
+		if ( withConstantScore ) {
+			result = applyConstantScore( result );
+		}
+
+		if ( !context.isExplicitNested() && nestedPathHierarchy != null && !nestedPathHierarchy.isEmpty() ) {
```

Comment:
I believe this could be more fine-grained: instead of checking whether there's explicit nesting, compare the closest nested predicate with the nested path hierarchy of this predicate.

For example, let's assume you have a field with two levels of nesting: `nested1.nested2.stringField`. If the user applies a nested predicate on `nested1` only, and forgets about `nested2` (which could be understandable in some situations) ... then we could add the nesting on `nested2` automatically when we detect the "closest nested predicate" doesn't match the last path in `nestedPathHierarchy`.

Reply:
I believe this comment still stands?

Reply:
I think that the test `ImplicitNestedSearchPredicateIT.javanested_X2_explicit_implicit` covers the case. Doesn't it?

Reply:
Looks like it. Thanks.